### PR TITLE
[JEX-136] Use modern `mixlib-install` to get packages from `packages.chef.io`.

### PR DESCRIPTION
**Pull request by yzl**
### Description

Install packages from `packages.chef.io` by updating to `mixlib-install ~> 1.0`.

### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

File: recipes/install.rb [hunk 1]
```diff
@@ -38,13 +35,14 @@
...
-  Chef::Log.info("['push_jobs']['package_url'] and ['push_jobs']['package_checksum'] not set. Chef Push client will be installed from CHEF's public repositories.")
+  Chef::Log.info("['push_jobs']['package_url'] and ['push_jobs']['package_checksum'] not set. Chef Push client will be installed from packages.chef.io.")
 end
 
 #
-# This uses the chef package repositories by default; a separate config step may be needed so that they are accepted.
-# That is superior to the "options '--force-yes'" clause that was used in one version.
-chef_ingredient 'push-client' do
+# This uses packages.chef.io by default.
+# TODO:  Tests are failing because the push client's config is dropped into /etc/chef, not /etc/push-jobs-client.
```
Comment (sersut) [inline recipes/install.rb]: Is this TODO still applicable?

**Comment by scotthain** [comment]
👍

**Comment by patrick-wright** [comment]
code looks good.  👍

**Comment by yzl** [comment]
cc @mmzyk

**Pull request merged**